packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -39,14 +38,15 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
   }
 
   Future<void> _isGapiInitialized;
+  Future<void> _isAuthInitialized = Completer<void>().future;
```

Comment:
seems odd to set `_isAuthInitialized` to a `Future` that will never complete. If someone does `await initialized;` before `init()` has been called, then they will be awaiting forever.

I suggest creating the `Completer<void>` for `_isAuthInitialized` in the constructor

Reply:
I was thinking about this. I don't like futures that never complete either.

I'll add some sort of boolean flag that I can quickly check (and throw a StateError or similar) when you call other methods before init, and THEN, have this as a proper completer that will be created at the right moment by the .init function.

That way the plugin can bail out early, if you call anything before init, and will be able to wait for init to be ready elsewhere.

Thanks for the nudge :P